apps/ai-image-generator/frontend/src/clients/segmentClient.ts

```diff
@@ -0,0 +1,53 @@
+import { config } from '../configs/config';
```

Comment:
What are your thoughts on using provider vs a singleton class here? If you use this code you'll need to change the `getUserCookieConsent` to accept BaseAppSDK instead of the deprecated KnownSDK

eg.
```
import { BaseAppSDK } from '@contentful/app-sdk';
import { AnalyticsBrowser } from '@segment/analytics-next';
import { config } from 'config';
import { createContext, ReactNode } from 'react';
import { getUserCookieConsent } from 'utils/cookieConsent';

interface SegmentAnalyticsContextProps {
  identify: (sdk: BaseAppSDK) => void;
  trackImageGenerationClick: (sdk: BaseAppSDK) => void;
  trackSelectAndEditClick: (sdk: BaseAppSDK) => void;
  trackImageRegenerationClick: (sdk: BaseAppSDK) => void;
  trackSelectAndEditRegenerationClick: (sdk: BaseAppSDK) => void;
}

const noop = () => {};

export const SegmentAnalyticsContext = createContext<SegmentAnalyticsContextProps>({
  identify: noop,
  trackImageGenerationClick: noop,
  trackSelectAndEditClick: noop,
  trackImageRegenerationClick: noop,
  trackSelectAndEditRegenerationClick: noop,
});

export const SegmentAnalyticsProvider = (props: { children: ReactNode }) => {
  const writeKey = config.segmentWriteKey;
  const segmentAnalytics = AnalyticsBrowser.load({ writeKey });

  const identify = (sdk: BaseAppSDK) => {
    if (getUserCookieConsent(sdk, 'PERSONALIZATION')) {
      segmentAnalytics.identify();
    }
  };

  const trackImageGenerationClick = (sdk: BaseAppSDK) => {
    if (getUserCookieConsent(sdk, 'ANALYTICS')) {
      segmentAnalytics.track('image_generation_click');
    }
  };

  const trackSelectAndEditClick = (sdk: BaseAppSDK) => {
    if (getUserCookieConsent(sdk, 'ANALYTICS')) {
      segmentAnalytics.track('select_and_edit_click');
    }
  };

  const trackImageRegenerationClick = (sdk: BaseAppSDK) => {
    if (getUserCookieConsent(sdk, 'ANALYTICS')) {
      segmentAnalytics.track('regenerate_image_generation');
    }
  };

  const trackSelectAndEditRegenerationClick = (sdk: BaseAppSDK) => {
    if (getUserCookieConsent(sdk, 'ANALYTICS')) {
      segmentAnalytics.track('regenerate_image_select_and_edit');
    }
  };

  return (
    <SegmentAnalyticsContext.Provider
      value={{
        identify,
        trackImageGenerationClick,
        trackSelectAndEditClick,
        trackImageRegenerationClick,
        trackSelectAndEditRegenerationClick,
      }}>
      {props.children}
    </SegmentAnalyticsContext.Provider>
  );
};
```